daemon/endpoint.go

```diff
@@ -177,6 +180,13 @@ func (d *Daemon) createEndpoint(epTemplate *models.EndpointChangeRequest, id str
 		}
 	}
 
+	// In order to capture initial "waiting-for-identity" state
+	// we need to increment the counters before `SetIdentityLabels`
+	// since `SetIdentityLabels` can trigger an endpoint regenerate
+	// and change the state to "waiting-to-regenerate"
+	metrics.EndpointStateCount.
```

Comment:
Since this is called in `createEndpoint` the comment states that in order to capture the initial state `waiting-for-identity` we need to call it before we set the identity.

Reply:
I can change the wording to more generic:
```
// In order to capture initial states
// we need to increment the counters before `SetIdentityLabels`
// since `SetIdentityLabels` can trigger an endpoint regenerate
// and change the state to "waiting-to-regenerate"
```